### PR TITLE
DT-362: added endpoint to expose Delius update case note endpoint

### PR DESCRIPTION
**Pull request by SimonMitchellMOJ**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/config/AlfrescoConfig.java [hunk 1]
```diff
@@ -9,12 +9,22 @@
...
-    @Bean
+    @Bean("alfrescoRestTemplate")
     public RestTemplate alfrescoRestTemplate(RestTemplateBuilder restTemplateBuilder,
                                              @Value("${alfresco.baseUrl}") String baseUrl) {
         return restTemplateBuilder
                 .rootUri(baseUrl)
                 .build();
     }
 
+    @Bean("alfrescoRestTemplateWithAuth")
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/delius/config/AlfrescoConfig.java]: alfresco with auth :confused:

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -70,6 +70,9 @@ delius:
   ldap:
     users:
       base: "cn=Users,dc=moj,dc=com"
+  baseurl: 'http://localhost:8088/alfresco/s/noms-spg'
+  username: 'A'
+  password: 'A'
```
Comment (petergphillips) [inline src/main/resources/application.yml]: don't think we should have defaults for this in the application.yml

File: src/main/java/uk/gov/justice/digital/delius/service/CaseNoteService.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+public class CaseNoteService {
+
+    private final RestTemplate restTemplate;
+
+    @Autowired
+    public CaseNoteService(@Qualifier("alfrescoRestTemplateWithAuth") RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public void upsertCaseNotesToDelius(final String nomisId, final Long caseNotesId, final String caseNote) {
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/delius/service/CaseNoteService.java]: Couldn't we model the object rather than a JSON string?

Comment (petergphillips) [reply src/main/java/uk/gov/justice/digital/delius/service/CaseNoteService.java]: is there any point?  We're just acting as a proxy here so don't really care what the data is...

File: src/main/resources/application.yml (hunk 2 above)
Comment (petergphillips) [reply src/main/resources/application.yml]: changed my mind based on the `dgoss` stuff running.  When that gets removed then we can remove this

File: src/main/java/uk/gov/justice/digital/delius/service/CaseNoteService.java (hunk 3 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/digital/delius/service/CaseNoteService.java]: Agree with Peter

**Pull request merged**